### PR TITLE
feat(console): phone number validation with flags

### LINKED ISSUE #3918: Phone number Validation behavior
As an admin or end user I want to be able to enter my phone no matter of the format without it being refactored while typing.

**Acceptance Criteria**
- [ ] I am able to enter spaces while entering my phone number
- [ ] I can enter 00XX as country prefix (it changes when I click save)
- [ ] I am able to enter 00 and a non swiss prefix
- [ ] The number is not formatted while I enter it
- [ ] The number is formatted when I click enter
- [ ] I am able to choose the country prefix with a dropdown

**Additional context**

see google contacts for a good example how to implement it (TL;DR: allow all inputs and after saving trunkate it to a coherent template (no spaces, country prefix with +, etc..)

The copy text is not very intuitive on how to enter the phone number (no example, not stating any format (i.e. country prefix)). this issue might get obsolete if the following things are improved:

Image

### PR DESCRIPTION
**Pull request by doncicuto**
As an admin or end user I want to be able to enter my phone no matter of the format without it being refactored while typing.

Acceptance Criteria

- [X] I am able to enter spaces while entering my phone number
- [X] I can enter 00XX as country prefix (it changes when I click save)
- [X] I am able to enter 00 and a non swiss prefix
- [X] The number is not formatted while I enter it
- [X] The number is formatted when I click enter
- [X] I am able to choose the country prefix with a dropdown

A discussion about design issues and a video showing proposed behavior is available at [https://github.com/zitadel/zitadel/discussions/5115]([https://github.com/zitadel/zitadel/discussions/5115)

closes #3918 

- [X] I am happy with the code
- [X] Short description of the feature/issue is added in the pr description
- [X] PR is linked to the corresponding user story
- [X] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [X] No debug or dead code
- [X] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [ ] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [X] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

**Review by peintnermax — CHANGES_REQUESTED** [review]
Nice work. Just found some minor things 👍

File: console/src/app/pages/users/user-create/user-create.component.ts [hunk 1]
```diff
@@ -190,7 +182,21 @@ export class UserCreateComponent implements OnDestroy {
     }
 
     if (this.phone && this.phone.value) {
-      humanReq.setPhone(new AddHumanUserRequest.Phone().setPhone(this.phone.value));
+      // Try to parse number and format it according to country
```
[the remark this answers is not in the document]
Comment (doncicuto) [reply console/src/app/pages/users/user-create/user-create.component.ts:185]: Absolutely @peintnermax

File: console/src/assets/i18n/it.json [hunk 2]
```diff
@@ -424,7 +425,8 @@
         "PREVIEW": "Anteprima",
         "DELETESUCCESS": "Rimosso con successo!",
         "CROPPERERROR": "È successo un errore durante il caricamento del tuo file. Prova un formato e una dimensione diversi, se necessario."
-      }
+      },
+      "COUNTRY": "Nazione"
```
[the remark this answers is not in the document]
Comment (doncicuto) [reply console/src/assets/i18n/it.json]: Changed!

File: console/src/assets/i18n/it.json [hunk 3]
```diff
@@ -405,6 +405,7 @@
       "TITLE": "Profilo",
       "EMAIL": "E-mail",
       "PHONE": "Numero di telefono",
+      "PHONE_HINT": "Utilizzare 00 o il simbolo + seguito dal prefisso del paese del chiamante, oppure selezionare il paese dall'elenco a discesa e infine inserire il numero di telefono",
```
[the remark this answers is not in the document]
Comment (doncicuto) [reply console/src/assets/i18n/it.json]: Changed!

File: console/src/assets/i18n/de.json [hunk 4]
```diff
@@ -405,6 +405,7 @@
       "TITLE": "Profil",
       "EMAIL": "E-Mail",
       "PHONE": "Telefonnummer",
+      "PHONE_HINT": "Verwenden Sie 00 oder das Symbol + gefolgt von der Landesvorwahl des Anrufers oder wählen Sie das Land aus der Dropdown-Liste aus und geben Sie schließlich die Telefonnummer ein",
```
[the remark this answers is not in the document]
Comment (doncicuto) [reply console/src/assets/i18n/de.json]: Changed!

File: console/src/app/pages/users/user-create/user-create.component.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (doncicuto) [reply console/src/app/pages/users/user-create/user-create.component.ts:186]: Now formatPhone is used in user-create.component.ts and edit-dialog.component.ts so code is now cleaner and reuses functionality

**Comment by juergrinaldi** [comment]
@doncicuto Thanks a lot for your contribution. We'd love to send you a little 'thank you' via mail. If you're up for it, you can fill out this form.
https://docs.google.com/forms/d/1FCzRnM4fMmteosNuc-g1TxjG5BZlXlfYAfF1xvl-vFA

**Review by peintnermax — APPROVED** [review]

**Pull request merged**